packages/mui-base/src/Button/Button.types.ts

```diff
@@ -30,6 +30,11 @@ export interface ButtonOwnProps extends Omit<UseButtonParameters, 'rootRef'> {
    * @default {}
    */
   slots?: ButtonSlots;
+  /**
+   * The HTML element, e.g.'button', 'a' etc
+   * @default 'button'
+   */
+  hostElementName?: keyof HTMLElementTagNameMap;
```

Comment:
I'm thinking about renaming this prop. "host element" is redundant (could be "host component", as React calls the native components), but to shorten it, I'd go with "elementName" or maybe "rootElementName" to be more explicit.

Reply:
@michaldudak `rootElementName` sounds good ~ I've renamed all the references